### PR TITLE
Add regions api

### PR DESCRIPTION
**Pull request by ndedic**
Implements Vault wrappers for Regions OpenStack API. It hasn't been tested properly on Vault side due to issues with running Vault server, but integration with OpenStack and tests should be working.
If this is roughly ok, I can look to implement Groups as well.

### CODE REVIEW

**Comment by ror6ax** [comment]
Thanks, looks awesome! If you want, you can start looking into Groups as well and meanwhile we'll figure out that problem with plugin versions. Also, pls rebase as 0.0.2 has been added.

**Comment by ndedic** [comment]
Thanks, I will look into Groups this week (and rebase).

**Comment by ndedic** [comment]
This should be ok now. Also, re: Groups, most of it seems to be already implemented. Did you have any specific method/requirement in mind?

File: plugin/path_regions.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	parentRegionID := data.Get("parent_region_id").(string)
+
+	// Store it
+	entry, err := logical.StorageEntryJSON("region/"+id, &regionEntry{
+		Region_id:               id,
+		Region_description:      description,
+		Region_parent_region_id: parentRegionID,
+	})
+
+	if err != nil {
```
Comment (ror6ax) [inline plugin/path_regions.go:122]: Seems you double check the err in next block.

File: plugin/regions_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/google/gofuzz"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRegions(t *testing.T) {
+
+	f := fuzz.New()
+	var id, description, parentRegionID string
+	f.Fuzz(&id)
+	//f.Fuzz(&parentRegionID)
```
Comment (ror6ax) [inline plugin/regions_test.go]: Probably forgot to uncomment.

**Review by ror6ax — COMMENTED** [review]
Little cleanup and we're good to go.

**Comment by ndedic** [comment]
Should be ok now.

File: plugin/path_regions.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.UpdateOperation: b.pathRegionWrite,
+			logical.ReadOperation:   b.pathRegionRead,
+		},
+	}
+}
+
+func (b *backend) Region(s logical.Storage, n string) (*regionEntry, error) {
+	entry, err := s.Get("region/" + n)
```
Comment (Trojan295) [inline plugin/path_regions.go:45]: I would recommend to refactor error checks to:
```
if entry, err := s.Get("region/" + n); err != nil {
  return nil, err
}
// continue function
```
You are using this syntax already in some places. Let's keep it consequent.

Comment (ndedic) [reply plugin/path_regions.go:45]: `entry` is scoped to `if` block but is required outside the block @ line 55... so this is just one way to do it...

**Pull request merged**